### PR TITLE
Allow extern static and GlobalAsm without verification support

### PR DESCRIPTION
**Pull request by ziqiaozhou**
Those features are important for writing bare-metal code in rust

```
global_asm!(
    "..."
);

extern "C" {
    #[verifier(external)]
    static extern_variable: u64;
}
```
NOTE: Rust compiler will throw away attribute for the builtin macro global_asm

### CODE REVIEW

**Review by Chris-Hawblitzel — APPROVED** [review]

**Comment by utaal** [comment]
Hi! Thanks for the PR.

Does this work for `extern "C"` functions too? (as long as they're marked `#[verifier(external)]`)?

Can you briefly explain how you'd be planning to use `global_asm` and why you'd prefer that over inline `asm!`?

**Comment by ziqiaozhou** [comment]
> 



> Hi! Thanks for the PR.
> 
> Does this work for `extern "C"` functions too? (as long as they're marked `#[verifier(external)]`)?
> 
> Can you briefly explain how you'd be planning to use `global_asm` and why you'd prefer that over inline `asm!`?

I was writing OS-level code for a AMD-SNP VM. In an early stage, I want to use global_asm to define the entry point so that I can initialize stack and then call verus-verified functions.

**Comment by utaal** [comment]
Thanks for the explanation; that's cool!

I'm not super comfortable with the idea of allowing global_asm anywhere and not require the user to explicitly allow (with, e.g., an attribute) it if it can influence the behavior of other code (which it probably can, if it can be the entry point).

If an attribute on `global_asm` itself doesn't work due to the builtin macro, what about a module level attribute? E.g.
`#![verifier(allow_global_asm)]`? I can look into how to make that work if you think that's reasonable.

File: source/rust_verify/src/rust_to_vir.rs [hunk 1]
```diff
@@ -425,6 +425,11 @@ fn check_item<'tcx>(
             // type alias (like lines of the form `type X = ...;`
             // Nothing to do here - we can rely on Rust's type resolution to handle these
         }
+        ItemKind::GlobalAsm(..) =>
+        //if get_verifier_attrs(ctxt.tcx.hir().attrs(item.hir_id()))?.external =>
```
Comment (utaal) [inline source/rust_verify/src/rust_to_vir.rs:429]: Please remove this comment and add a TODO to add a crate-level attribute to enable `global_asm`.

**Review by utaal — APPROVED** [review]

File: source/rust_verify/src/rust_to_vir.rs (hunk 1 above)
Comment (ziqiaozhou) [reply source/rust_verify/src/rust_to_vir.rs]: Fixed this comment.

**Pull request merged**